lib/console/deployer/operations.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Console.Deployer.Operations do
+  alias Console.Schema.Build
+  alias Console.Commands.{Plural}
+  alias Console.Plural.Context
+
+  def perform(storage, %Build{repository: repo, type: :bounce} = build) do
+    with_build(build, [
+      {storage, :init, []},
+      {Plural, :bounce, [repo]}
+    ], storage)
+  end
+
+  def perform(storage, %Build{type: :deploy, repository: repo, message: message} = build) do
+    with_build(build, [
+      {storage, :init, []},
+      {Plural, :build, [repo]},
+      {Plural, :diff, [repo]},
+      {Plural, :deploy, [repo]},
+      {storage, :revise, [commit_message(message, repo)]},
+      {storage, :push, []}
+    ], storage)
+  end
+
+  def perform(storage, %Build{type: :install, context: %{"configuration" => conf, "bundle" => b}, message: message} = build) do
+    with_build(build, [
+      {storage, :init, []},
+      {Context, :merge, [conf, %Context.Bundle{repository: b["repository"], name: b["name"]}]},
+      {Plural, :build, []},
+      {Plural, :install, [b["repository"]]},
+      {storage, :revise, [commit_message(message, b["repository"])]},
+      {storage, :push, []}
+    ], storage)
+  end
+
+  def perform(storage, %Build{type: :install, context: %{"configuration" => conf, "bundles" => bs}, message: message} = build) do
+    with_build(build, [
+      {storage, :init, []},
+      {Context, :merge, [conf, Enum.map(bs, fn b -> %Context.Bundle{repository: b["repository"], name: b["name"]} end)]},
+      {Plural, :build, []},
+      {Plural, :install, [Enum.map(bs, & &1["repository"])]},
+      {storage, :revise, [message]},
+      {storage, :push, []}
+    ], storage)
+  end
+
+  def perform(storage, %Build{type: :approval, repository: repo, message: message} = build) do
+    with_build(build, [
+      {storage, :init, []},
+      {Plural, :build, [repo]},
+      {Plural, :diff, [repo]},
+      :approval,
+      {Plural, :deploy, [repo]},
+      {storage, :revise, [commit_message(message, repo)]},
+      {storage, :push, []}
+    ], storage)
+  end
+
+  defp perform(storage, %Build{type: :destroy, repository: repo} = build) do
```

Comment:
@michaeljguarino There is a compiler error in this line.

Reply:
ugh bad visual merge, will fix in a bit